### ISSUE TITLE
How to close Googlebot (recaptcha)

### ISSUE THREAD

**Issue by appmlk** [overleaf/overleaf#652]
I can not connect to Google in China
So OverLeaf installed in my server is very slow
How to close recaptcha？o(╥﹏╥)o

**Comment by mserranom** [comment]
Sorry to hear this, please contact our customer support via https://www.overleaf.com/contact.

**Comment by mserranom** [comment]
Reopening and renaming, I misunderstood that the problems were related to overleaf.com instead of Overleaf Community Edition.

**Comment by das7pad** [comment]
You need to overwrite the `recaptcha`  field in the settings. Actually it should be disabled by default, as there is no need to enable it for local development work either.

```coffee
# settings.coffee
module.exports =
  # ...
  recaptcha: undefined
  # ...
```

**Comment by ShaneKilkelly** [comment]
It's my understanding that recaptcha should be disabled by default:
https://github.com/overleaf/web/blob/master/config/settings.defaults.coffee#L475

@appmlk can you give an example of a case where recaptcha is active and causing trouble? Thanks!

**Comment by das7pad** [comment]
@ShaneKilkelly it's loaded here:

https://github.com/overleaf/web/blob/449ad181df66d789bb1f66ca8b8e31a2713a7a9c/app/views/layout.pug#L88

**Comment by ShaneKilkelly** [comment]
Ah, and if those endpoints are blocked it'll slow the loading of the page? I'll open an internal ticket to figure this out.

**Comment by das7pad** [comment]
It's not just a matter of performance, but also a waste of bandwidth and more importantly privacy concerns. I simply do not see a need to grant Google deep insights into every ones development process or private overleaf deployment - after all they have FULL access on the dom/events/...

**Comment by ShaneKilkelly** [comment]
Yup, totally agree. We've opened an internal ticket to fix this :+1:

**Comment by das7pad** [comment]
landed in https://github.com/overleaf/web/commit/5aa830df0339d1b657c92b8585fb0a546224ae71

**Comment by HPDell** [comment]
Is the reCAPTCHA now disabled by default? But I found it is still need to load it after deploying the community edition yesterday. 

How to disable it now?

**Comment by mserranom** [comment]
Change https://github.com/overleaf/web/commit/5aa830df0339d1b657c92b8585fb0a546224ae71 was rolled back since it was causing a separate issue, reCAPTCHA is still enabled in all cases.

**Comment by HPDell** [comment]
I found that if I set `settings.recaptcha = false` in /etc/sharelatex/settings.coeff, I will not able to send invite email. How to fix it?

**Comment by ShaneKilkelly** [comment]
We're currently working on a fix for this issue internally, sorry for the delay!

**Comment by TJJTJJTJJ** [comment]
So, is there some way to disable reCAPTCHA when OverLeaf installed in my server is very slow? @ShaneKilkelly

**Comment by TJJTJJTJJ** [comment]
I found one way to disable recaptcha.
In docker,
```bash
docker exec -it sharelatex bash
vim /var/www/sharelatex/web/config/settings.defaults.coffee

    472         # recaptcha:
    473         #       disabled:
    474         #               invite: true
    475         #               register: true

exit
docker restart sharelatex
```
.

In my opinion, images: sharelatex/sharelatex use old web, not the newest web. https://github.com/overleaf/web/blob/master/config/settings.defaults.coffee#L475

**Comment by das7pad** [comment]
Hello,

Thanks for the report. reCAPTCHA has been disabled by default in Overleaf Community edition and [Server Pro](https://www.overleaf.com/for/enterprises) for a few major releases now.

Greetings,
Jakob
